lib/Sema/CSApply.cpp

```diff
@@ -4112,17 +4096,13 @@ namespace {
       if (hasForcedOptionalResult(expr))
         toType = toType->getOptionalObjectType();
 
-      auto castContextKind = SuppressDiagnostics || expr->isImplicit()
-                                 ? CheckedCastContextKind::None
-                                 : CheckedCastContextKind::ForcedCast;
-
       const auto castKind = TypeChecker::typeCheckCheckedCast(
-          fromType, toType, castContextKind, dc, expr->getLoc(), sub,
-          castTypeRange);
+          fromType, toType, CheckedCastContextKind::ForcedCast, dc);
       switch (castKind) {
         /// Invalid cast.
       case CheckedCastKind::Unresolved:
-        return nullptr;
```

Comment:
This is interesting, isn't `Unresolved` here an actual error condition?

Reply:
This is indeed a good point, in the original logic, for `CheckedCastContextKind` that were not `None` or `Coercion` every failure would here be diagnosed returned as a `ValueCast` otherwise `Unresolved` would be the result. Now that no diagnostic is being emitted the failures only `Unresolved` which is not the expected for later stages where diagnostics were emitted (which seems to expect a cast kind in the expr). I changed so the condition where `Unresolved` (expr->isImplicit()) we continue to return a failure, otherwise set as value cast and continue. Not sure if we should account for the `SuppressDiagnostics` flag in this case.

Reply:
I think regardless of `SuppressDiagnostics` it would now behave consistently with previous version so it's okay.

Reply:
Ah awesome!